src/System.Private.CoreLib/shared/System/Runtime/Serialization/SafeSerializationEventArgs.cs

```diff
@@ -6,24 +6,13 @@
 
 namespace System.Runtime.Serialization
 {
-    // SafeSerializationEventArgs are provided to the delegates which do safe serialization.  Each delegate
-    // serializes its own state into an IDeserializationCallback instance which must, itself, be serializable.
-    // These indivdiual states are then added to the SafeSerializationEventArgs in order to be saved away when
-    // the original ISerializable type is serialized.
+    // This type exists for public surface compatibility only.
     public sealed class SafeSerializationEventArgs : EventArgs
     {
-        private readonly List<object> _serializedStates = new List<object>();
-
-        internal SafeSerializationEventArgs() { }
+        private SafeSerializationEventArgs() { }
 
         public void AddSerializedState(ISafeSerializationData serializedState)
         {
-            if (serializedState == null)
-                throw new ArgumentNullException(nameof(serializedState));
-            if (!serializedState.GetType().IsSerializable)
-                throw new ArgumentException(SR.Format(SR.Serialization_NonSerType, serializedState.GetType(), serializedState.GetType().Assembly.FullName));
```

Comment:
Are any corefx tests going to need to be changed/removed?

Reply:
This was unreachable code. It has not been possible to legally instantiate this type (except for private reflection).

Reply:
Ah, yup, thanks.